pkg/cmd/grafana-cli/commands/datamigrations/encrypt-datasource-passwords.go

```diff
@@ -0,0 +1,100 @@
+package datamigrations
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/utils"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/grafana/grafana/pkg/util/errutil"
+)
+
+func EncryptDatasourcePaswords(c utils.CommandLine, sqlStore *sqlstore.SqlStore) error {
+	var passwordRows []map[string]string
+	datasourceTypes := []string{
+		"mysql",
+		"influxdb",
+		"elasticsearch",
+		"graphite",
+		"prometheus",
+		"opentsdb",
+	}
+	return sqlStore.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		sess.Cols("id", "password", "secure_json_data")
+		sess.Table("data_source")
+		sess.In("type", datasourceTypes)
+		sess.Where("password IS NOT NULL AND password != ''")
```

Comment:
Due to GOs zero values we do not really have NULLs in DB so that part could be omitted but it is probably better to be sure.